File: leetcode/125/125.cpp
#include <iostream>
#include <ctype.h>
#include <string>
#include <cstdio>
using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        int strSize = s.length();
        string str;
        for(int i =0; i < strSize; i++)
        {
            if(isalnum(s[i]))
            {
                str.push_back(tolower(s[i]));
            }
        }
        int newStrSize = str.length();
        for(int i =0; i < newStrSize/2; i++)
        {
            if(str[i] != str[newStrSize -1 - i])
            {
                return false;
            }
        }
        return true;
    }
};


int main()
{
    Solution s;
    printf("%d\n", s.isPalindrome("A man, a plan, a canal: Panama"));
}

File: leetcode/125/125.py
""" 
class Solution:
    def isPalindrome(self, s: str) -> bool:
        str = []
        for c in s:
            if c.isalnum():
                str.append(c.lower())
        print(str)
        while len(str) > 1:
            if str.pop(0) != str.pop():
                return False
        return True
"""
class Solution:
    def isPalindrome(self, s: str) -> bool:
        str = []
        for c in s:
            if c.isalnum():
                str.append(c.lower())
        if str != str[::-1]:
            return False
        return True

## 리스트 슬라이싱이 제일 빠르다
